lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -454,15 +455,16 @@ def add_response_waiter(request, completion_routine = nil, completion_param = ni
   # if anyone.
   #
   def notify_response_waiter(response)
+    handled = false
     self.waiters.each() { |waiter|
       if (waiter.waiting_for?(response))
         waiter.notify(response)
-
         remove_response_waiter(waiter)
-
+        handled = true
```

Comment:
For me a `return true` is a bit clearer than `break` as to the logic (ie notify the first waiter we find), but its just preferences.